File: golang/code_examples.go
package main

import (
    "bufio"
    "encoding/json"
    "fmt"
    "net/http"
    "os"
    "strings"
)

type AuthResponse struct {
    Auth     bool   `json:"auth"`
    Username string `json:"username"`
}

func readOrCreateKey() (string, error) {
    keyPath := "key.dat"
    var key string
    if file, err := os.Open(keyPath); err == nil {
        defer file.Close()
        scanner := bufio.NewScanner(file)
        if scanner.Scan() {
            key = scanner.Text()
        }
    } else {
        fmt.Print("Enter your auth key: ")
        scanner := bufio.NewScanner(os.Stdin)
        if scanner.Scan() {
            key = scanner.Text()
            os.WriteFile(keyPath, []byte(key), 0644)
        }
    }
    return key, nil
}

func getHWID() string {
    return "SIMULATED_HWID" // Replace with actual HWID retrieval logic.
}

func authenticate(key string) {
    hwid := getHWID()
    resp, err := http.PostForm("https://cracked.to/auth.php",
        url.Values{"a": {"auth"}, "k": {key}, "hwid": {hwid}})
    if err != nil {
        fmt.Println("Failed to authenticate:", err)
        return
    }
    defer resp.Body.Close()

    var response AuthResponse
    json.NewDecoder(resp.Body).Decode(&response)
    if response.Auth {
        fmt.Println("Welcome, " + response.Username)
    } else {
        fmt.Println("Access denied.")
    }
}

func main() {
    key, err := readOrCreateKey()
    if err != nil {
        fmt.Println("Error:", err)
        return
    }
    authenticate(key)
}

File: rust/code_examples.rs
```rust
use reqwest::Client;
use serde::{Deserialize, Serialize};
use std::{fs, io::{self, Write}, path::Path};

#[derive(Serialize, Deserialize)]
struct AuthResponse {
    auth: bool,
    username: Option<String>,
}

async fn read_or_create_key() -> io::Result<String> {
    let key_path = Path::new("key.dat");
    match fs::read_to_string(key_path) {
        Ok(key) => Ok(key),
        Err(_) => {
            print!("Enter your auth key: ");
            io::stdout().flush()?;
            let mut key = String::new();
            io::stdin().read_line(&mut key)?;
            fs::write(key_path, &key)?;
            Ok(key.trim().to_string())
        }
    }
}

async fn get_hwid() -> String {
    "SIMULATED_HWID".to_string() // Replace with actual HWID retrieval logic.
}

async fn authenticate(key: String) -> Result<(), Box<dyn std::error::Error>> {
    let client = Client::new();
    let hwid = get_hwid().await;
    let res = client.post("https://cracked.to/auth.php")
        .form(&[("a", "auth"), ("k", &key), ("hwid", &hwid)])
        .send()
        .await?;

    let response: AuthResponse = res.json().await?;
    if response.auth {
        println!("Welcome, {}", response.username.unwrap_or_default());
    } else {
        println!("Access denied.");
    }

    Ok(())
}

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let key = read_or_create_key().await?;
    authenticate(key).await
}
```
